src/styles/utils/zIndexStyle.tsx

```diff
@@ -0,0 +1,29 @@
+import { css } from "styled-components";
+
+import { ResponsiveValues, responsiveStyle } from "./responsiveStyle";
+
+export const zIndexMap = {
+  behind: -1,
+  neutral: 0,
+  inFront: 1,
+  mobileFilters: 2,
+  fixedHeader: 100,
+  modalCloseButton: 150,
+  modalDialog: 300,
+} as const;
+
+export type ZIndex = keyof typeof zIndexMap | null;
+export type ZIndexProps = {
+  $zIndex?: ResponsiveValues<ZIndex>;
+};
+
+export const parseZIndex = (value?: ZIndex | null) => {
```

Comment:
I don't think these parse functions should be here.  We could actually do this without the function simply by improving the default callback in responsiveStyle.ts . Take a look at line 54 of that file and see if you can do this whilst removing this one.

Reply:
hmm the reason we need the parse functions is whenthe props take semantic names which need parsing to css values by the responsiveStyle function. if we dont parse the parse function into responsiveStyles then they won't know which table to cross reference to select correct css value.